### ISSUE TITLE
Refactor `dependents` method

### ISSUE THREAD

**Issue by doowb** [doowb/npm-api-dependents#1]
I'd like to refactor this library, which is mainly the main `dependents` method, so it's easier to use downstream as a streaming module instead of batching reads.

- [ ] Remove generator syntax and use promises directly when necessary
- [ ] use `view.stream` instead of `view.query`
- [ ] `options.mapFn` won't be necessary. Since this is an `npm-api` plugin, values can be transformed directly into `Repo` objects using `app.repo(name)` then pushed onto the stream.
- [ ] `options.raw` consider adding `options.raw` to push data directly through the stream without initializing new `Repo` instances.

**Comment by ralphtheninja** [comment]
@doowb I've started looking at this. I'll hack something up and propose a PR for further discussion.

**Comment by ralphtheninja** [comment]
As a side note. I ran `examples/co.js` to make sure the data I received was the same when streaming and noticed that `3.0.0` is a dependent, which is odd?

![3 0 0-dependent](https://user-images.githubusercontent.com/308049/34363777-5d8ba9ec-ea7f-11e7-8f9c-411656a3c8c5.png)

**Comment by ralphtheninja** [comment]
Also, the total amount of dependents for `micromatch` I get from `co.js` is 264, but on `npmjs.com` it says 327, see https://www.npmjs.com/package/micromatch

**Comment by doowb** [comment]
The results are coming from this query: https://skimdb.npmjs.com/registry/_design/app/_view/dependedUpon?group_level=2&startkey=[%22micromatch%22]&endkey=[%22micromatch%22,{}]

I thought that was the same couchdb view that npm is using to display dependents on the site, but something may have changed.

**Comment by doowb** [comment]
>  noticed that 3.0.0 is a dependent, which is odd?

lol... that's an actual package in npm

**Comment by doowb** [comment]
NPM has another registry called `replicate` that works just like `skimdb` but also includes scoped packages. But scoped packages don't work with `registry.npmjs.com/{package}` :(

There are a couple of changes I can make in `npm-api` to use replicate and work around the other limitation. This will get the number close to what's on the npmjs.com website, but it's still not the same. I'm not sure where the extra dependencies are coming from on the website.

**Comment by DanielRuf** [comment]
> But scoped packages don't work with `registry.npmjs.com/{package}` :(
> 
> There are a couple of changes I can make in `npm-api` to use replicate and work around the other limitation. This will get the number close to what's on the npmjs.com website, but it's still not the same. I'm not sure where the extra dependencies are coming from on the website.

Should be the scoped (private) ones.

> But scoped packages don't work with `registry.npmjs.com/{package}`

In which way exactly?

**Comment by doowb** [comment]
> In which way exactly?

I don't remember why I mentioned that but the `registry.npmjs.com/{scope}/{package}` does return information now. I may have been thinking of the `downloads` services, but I'm not sure.

**Comment by DanielRuf** [comment]
Sounds like we can revisit this and try to add an option to include the scoped packages probably.

**Comment by doowb** [comment]
The URLs specifying the registry and "couchdb" instance to use are specified in the [npm-api package](https://github.com/doowb/npm-api/blob/2204395a7da8815465c12ba147cd7622b9d5bae2/lib/config.js#L11). I'm willing to take PRs to update or make the URL more configurable to allow handling different ones.

At once point I updated the URL to the `replicate` one, but I think there were issues with the service at the time and I switched it back to `skimdb`. The `registry.npmjs.com` is used for `package.json` data, which isn't used directly in this library (which is why I don't know why I brought it up before).
